### DISCUSSION TITLE
Osu android

### DISCUSSION THREAD

**Discussion by Rystars** [ppy/osu#18336] [Q&A]
how do you make an osu beatmap on android?
while osu android can't open its external storage, and can only open root storage, but my cellphone is not rooted

**Comment by frenzibyte** [comment]
Do you mean exporting the beatmap? It's not feasible in mobile for the time being. Otherwise not sure what limitation you're facing here.

**Reply by Hexality** [reply]
if he's talking about `/sdcard/Android/data` (or obb), those folders got locked up from android 11 and upwards, unrooted users doesn't have access to these folders anymore, and osu!lazer's export, rulesets and other folders are stored in there inside `/sdcard/Android/data/sh.ppy.osulazer/`

**Reply by Rystars** [reply]
> if he's talking about `/sdcard/Android/data` (or obb), those folders got locked up from android 11 and upwards, unrooted users doesn't have access to these folders anymore, and osu!lazer's export, rulesets and other folders are stored in there inside `/sdcard/Android/data/sh.ppy.osulazer/`

Yes, I let about that

**Comment by Hexality** [comment]
![image](https://user-images.githubusercontent.com/17398632/169424190-2a35a23b-1cf3-4562-b1a1-3ebdfa49047d.png)
![image](https://user-images.githubusercontent.com/17398632/169424200-216a0705-58f2-497a-bc51-176d1eb029c6.png)

**Reply by Hexality** [reply]
also, if you mean make a beatmap and then export it, android data folder got locked on android 11 and over so idk, maybe peppy will fix it in the future (?)

**Reply by Rystars** [reply]
> also, if you mean make a beatmap and then export it, android data folder got locked on android 11 and over so idk, maybe peppy will fix it in the future (?)

I hope Peppy fixes this




My phone is android 10

**Reply by Hexality** [reply]
weird, I don't remember android 10 having data folder locked lol

**Reply by frenzibyte** [reply]
As I mentioned above, mobile platforms are not allowed to export beatmaps right now, not necessarily due to hardware limitation.

**Reply by Rystars** [reply]
> weird, I don't remember android 10 having data folder locked lol

Mobile platforms are not allowed to open storage on osu.

**Reply by Hexality** [reply]
older than 11, yeah you can.
lazer itself doesn't allow you to access skin/song files, it's not just a mobile platform thing.

**Reply by Rystars** [reply]
Hopefully it will be fixed in the future
